apps/mongodb.go

```diff
@@ -0,0 +1,291 @@
+	c := []fluentbit.Component{}
+
+	jsonParser := &confgenerator.LoggingProcessorJsonParser{
+		ParserShared: confgenerator.ParserShared{
+			TimeKey:    "time",
+			TimeFormat: "%Y-%m-%dT%H:%M:%S.%L%z",
+			Types: map[string]string{
+				"id":      "integer",
+				"message": "string",
+			},
+		},
+	}
+	jpComponents := jsonParser.Components(tag, uid)
+
+	parserComponent, filterComponent := jpComponents[0], jpComponents[1]
+
+	c = append(c, parserComponent, filterComponent)
+
+	tempPrefix := "temp_ts_"
+	timeKey := "time"
+	// have to bring $date to top level in order for it to be parsed as timeKey
+	// see https://github.com/fluent/fluent-bit/issues/1013
+	liftTs := fluentbit.Component{
+		Kind: "FILTER",
+		Config: map[string]string{
+			"Name":         "nest",
+			"Match":        tag,
+			"Operation":    "lift",
+			"Nested_under": "t",
+			"Add_prefix":   tempPrefix,
+		},
+	}
+
+	renameTsOption := fluentbit.NewHardRenameModifyOptions(fmt.Sprintf("%s$date", tempPrefix), timeKey, nil)
+	renameTs := renameTsOption.Component(tag)
+
+	c = append(c, liftTs, renameTs)
+
+	// IMPORTANT: now that we have lifted the json to top level
+	// we need to re-parse in order to properly set time at the
+	// parser level
+	c = append(c, filterComponent)
+
+	removeTimestamp := fluentbit.Component{
+		Kind: "FILTER",
+		Config: map[string]string{
+			"Name":   "modify",
+			"Match":  tag,
+			"Remove": timeKey,
+		},
+	}
+	c = append(c, removeTimestamp)
+
+	return c
+}
+
+func (p *LoggingProcessorMongodb) severityParser(tag, uid string) []fluentbit.Component {
+	severityComponents := []fluentbit.Component{}
+	severityKey := "logging.googleapis.com/severity"
+
+	severityComponents = append(severityComponents, fluentbit.Component{
+		Kind: "FILTER",
+		Config: map[string]string{
+			"Match":  tag,
+			"Name":   "modify",
+			"Rename": "s severity",
+		},
+	})
+
+	severityComponents = append(severityComponents, fluentbit.TranslationComponents(tag, "severity", severityKey, []struct {
+		SrcVal  string
+		DestVal string
+	}{
+		{
+			SrcVal:  "D",
+			DestVal: "DEBUG",
+		},
+		{
+			SrcVal:  "D1",
+			DestVal: "DEBUG",
+		},
+		{
+			SrcVal:  "D2",
+			DestVal: "DEBUG",
+		},
+		{
+			SrcVal:  "D3",
+			DestVal: "DEBUG",
+		},
+		{
+			SrcVal:  "D4",
+			DestVal: "DEBUG",
+		},
+		{
+			SrcVal:  "D5",
+			DestVal: "DEBUG",
+		},
+		{
+			SrcVal:  "I",
+			DestVal: "INFO",
+		},
+		{
+			SrcVal:  "E",
+			DestVal: "ERROR",
+		},
+		{
+			SrcVal:  "F",
+			DestVal: "FATAL",
+		},
+		{
+			SrcVal:  "W",
+			DestVal: "WARNING",
+		},
+	})...)
+
+	severityComponents = append(severityComponents, fluentbit.Component{
+		Kind: "FILTER",
+		Config: map[string]string{
+			"Name":   "modify",
+			"Match":  tag,
+			"Remove": "severity",
+		},
+	})
+
+	return severityComponents
+}
+
+func (p *LoggingProcessorMongodb) renames(tag, uid string) []fluentbit.Component {
+	r := []fluentbit.Component{}
+	renames := []struct {
+		src  string
+		dest string
+	}{
+		{
```

Comment:
Same here. We could compress them like https://github.com/GoogleCloudPlatform/ops-agent/blob/d88a768910a7f4eb2a4bbae94bb829ea9effa261/apps/nginx.go#L146-L158 for space saving and readability.

Reply:
These are for a single field rename though kind of like a mapping right? Like for that one it only looking at the source key `level` and the moving to `logging.googleapis.com/severity`. I guess I could create some more analogous helper function but opted not to use the `TranslationComponents` because these are renaming multiple fields to be consistent. 

I also didn't want to break that function by chance since a lot of integrations currently rely on it, but if you can think of a way I would be interested!